test/cctest/test_environment.cc

```diff
@@ -0,0 +1,107 @@
+#include "node.h"
+#include "env.h"
+#include "v8.h"
+#include "libplatform/libplatform.h"
+
+#include <string>
+#include "gtest/gtest.h"
+#include "node_test_fixture.h"
+
+using node::Environment;
+using node::IsolateData;
+using node::CreateIsolateData;
+using node::FreeIsolateData;
+using node::CreateEnvironment;
+using node::FreeEnvironment;
+using node::AtExit;
+using node::RunAtExit;
+
+static bool called_cb_1 = false;
+static bool called_cb_2 = false;
+static void at_exit_callback1(void* arg);
+static void at_exit_callback2(void* arg);
+static std::string cb_1_arg;  // NOLINT(runtime/string)
+
+class EnvironmentTest : public NodeTestFixture {
+ public:
+  class Env {
+   public:
+    Env(v8::Isolate* isolate, char** argv) {
+      context_ = v8::Context::New(isolate);
+      isolate_data_ = CreateIsolateData(isolate, uv_default_loop());
+      environment_ = CreateEnvironment(isolate_data_,
+                                       context_,
+                                       1, argv,
+                                       2, argv);
```

Comment:
This is so-so.  There is nothing that enforces that `argv` is at least two elements big.